CMakeLists.txt

```diff
@@ -346,10 +347,16 @@ endif()
 set(CMAKE_FIND_FRAMEWORK LAST)
 
 if(UNIX)
+    set(AVIF_PLATFORM_LIBRARIES ${AVIF_PLATFORM_LIBRARIES} m)
+endif()
+
+if(AVIF_USE_THREAD)
     # Find out if we have threading available
-    set(THREADS_PREFER_PTHREAD_FLAG ON)
-    find_package(Threads)
-    set(AVIF_PLATFORM_LIBRARIES ${AVIF_PLATFORM_LIBRARIES} m Threads::Threads)
+    find_package(Threads QUIET)
```

Comment:
[On MINGW, this has potentially different behaviour depending on the presence of `winpthreads`](https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/FindThreads.cmake):

The target will be available always, and `CMAKE_USE_WIN32_THREADS_INIT` will always be defined. However, if `winpthreads` is also installed, `CMAKE_THREAD_LIBS_INIT` will also not be empty and `-lpthread` or `-pthread` will be added to the target unnecessarily. (I guess the linker might be smart enough to figure out there are no symbols used though and ignore it?)

Reply:
I'd says so yes.

Reply:
I just checked this - in the presence of `pthread.h` from `winpthreads`, the find module actually succeeds compiling the test code and (wrongly?) sets `CMAKE_HAVE_LIBC_PTHREAD` so `CMAKE_THREAD_LIBS_INIT` is still empty as a lucky side effect on MINGW:
```
-- CMAKE_THREAD_LIBS_INIT:
-- CMAKE_USE_WIN32_THREADS_INIT:1
-- CMAKE_USE_PTHREADS_INIT:1
```

Reply:
Just for completeness: although not affecting libavif because of the native only Win32 implantation, CMake's detection is still not 100% correct on MINGW:

https://gitlab.kitware.com/cmake/cmake/-/issues/6399
https://gitlab.kitware.com/cmake/cmake/-/merge_requests/7179

If `winpthreads` is present and preferred, it should be adding `-pthread` or non-empty `CMAKE_THREAD_LIBS_INIT` to the imported target.